backend/services/courses/term.py

```diff
@@ -0,0 +1,159 @@
+    ):
+        """Initializes the database session."""
+        self._session = session
+        self._permission = permission
+
+    def current(self) -> TermDetails:
+        """Gets the current term from the table.
+
+        Returns:
+            TermDetails: Currently active term.
+        """
+        # Select all entries in the `Term` table and sort by end date
+        query = select(TermEntity).order_by(TermEntity.end.desc())
+        entity = self._session.scalars(query).one_or_none()
+
+        # Raise an error if no entity was found.
+        if entity is None:
+            raise ResourceNotFoundException(f"There is no currently active term.")
+
+        # Return the model
+        return entity.to_details_model()
+
+    def all(self) -> list[TermDetails]:
+        """Retrieves all terms from the table
+
+        Returns:
+            list[TermDetails]: List of all `TermDetails`
+        """
+        # Select all entries in `Organization` table
+        query = select(TermEntity)
+
+        entities = self._session.scalars(query).all()
+
+        # Convert entries to a model and return
+        return [entity.to_details_model() for entity in entities]
+
+    def get_from_id(self, id: str) -> TermDetails:
+        """Gets the term from the table for an id.
+
+        Returns:
+            TermDetails: Term based on the id.
+        """
+        # Select all entries in the `Term` table and sort by end date
+        query = select(TermEntity).filter(TermEntity.id == id)
+        entity = self._session.scalars(query).one_or_none()
+
+        # Raise an error if no entity was found.
+        if entity is None:
+            raise ResourceNotFoundException(f"Term with id: {id} does not exist.")
+
+        # Return the model
+        return entity.to_details_model()
+
+    def create(self, subject: User, term: Term) -> Term:
+        """Creates a new term.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            term (Term): Term to add to table
+
+        Returns:
+            Term: Object added to table
+        """
+
+        # Check if user has admin permissions
+        self._permission.enforce(subject, "course.admin", f"term")
+
+        # Create new object
+        term_entity = TermEntity.from_model(term)
+
+        # Add new object to table and commit changes
+        self._session.add(term_entity)
+        self._session.commit()
+
+        # Return added object
+        return term_entity.to_model()
+
+    def update(self, subject: User, term: Term) -> Term:
+        """Updates a term.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            term (Term): Term to update
+
+        Returns:
+            Term: Object updated in the table
+        """
+
+        # Check if user has admin permissions
+        self._permission.enforce(subject, "course.admin", f"term")
```

Comment:
Permission `"courses.term.update"`, `"term/{term_id}"`

This is a bit pedantic because we'll never be giving term level permissions, but for consistency...